### PR TITLE
allow to disable link detection

### LINKED ISSUE #52: allow to disable automatic link detection
I moving that over from the original flow router repo:

https://github.com/kadirahq/flow-router/issues/705

I try to update to this version here, but my monkey patch from the original flow router no longer works, because _page is now hidden behind a getter :-(

So we need to adjust that in the initialisation code. 

Maybe i can do a PR for that

### PR DESCRIPTION
**Pull request by macrozone**
fixes https://github.com/VeliovGroup/flow-router/issues/52

- [x] decide on option name
- [x] mention in readme or docs
- [ ] a test?

### CODE REVIEW

**Comment by macrozone** [comment]
@dr-dimitru what do you think?

(it seems to work for me! but will test it a bit more)

File: docs/api/initialize.md [hunk 1]
```diff
@@ -1,14 +1,22 @@
 ### initialize method
 
 ```js
-FlowRouter.initialize();
+FlowRouter.initialize(options);
 ```
  - Returns {*void*}
```
Comment (dr-dimitru) [inline docs/api/initialize.md]: @macrozone need to list and define all arguments above `return`, as example take a [look here](https://github.com/VeliovGroup/flow-router/blob/master/docs/api/getParam.md)

File: client/router.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ class Router {
       const paramArr = param.split('');
       let _param   = '';
       for (let i = 0; i < paramArr.length; i++) {
-        if (!!~this._specialChars.indexOf(paramArr[i])){
+        if (~this._specialChars.indexOf(paramArr[i])){
```
Comment (dr-dimitru) [inline client/router.js]: Better to use `.includes()` *Array* method, or keep double negative.

File: client/router.js [hunk 3]
```diff
@@ -148,11 +148,11 @@ class Router {
       const queryParams = this._qs.parse(context.querystring);
       this._current = {
         path: context.path,
-        context: context,
+        context,
```
Comment (dr-dimitru) [inline client/router.js]: 👍

File: client/router.js [hunk 4]
```diff
@@ -376,15 +374,11 @@ class Router {
       subscriptions = _.values(globalRoute.getAllSubscriptions());
       subscriptions = subscriptions.concat(_.values(currentRoute.getAllSubscriptions()));
     } else {
-      subscriptions = _.map(args, (subName) => {
-        return globalRoute.getSubscription(subName) || currentRoute.getSubscription(subName);
-      });
+      subscriptions = _.map(args, (subName) => globalRoute.getSubscription(subName) || currentRoute.getSubscription(subName));
```
Comment (dr-dimitru) [inline client/router.js]: not really fan of one-liners, same for line :381 , is there any real benefits behind it?

File: client/router.js [hunk 5]
```diff
@@ -444,7 +438,8 @@ class Router {
     this._page.base(this._basePath);
     this._page({
```
Comment (dr-dimitru) [inline client/router.js]: I still believe we should pass `options.page` *Object* with all options for `page` package. Like:
```js
const pageOptions = Object.assign({
  hashbang: !!options.hashbang,
  decodeURLComponents: true
}, options.page);

this._page(pageOptions);
```

**Review by dr-dimitru — CHANGES_REQUESTED** [review]
@macrozone left some notes on docs and code. We should make it tweakable as much as possible, now there is not a lot of options for page package, but page may update their API and etc. so, it's better to give an option to tune it.

File: client/router.js (hunk 2 above)
Comment (macrozone) [reply client/router.js]: sorry, that was my linter/prettier probably. I thought i turned that off

File: client/router.js (hunk 3 above)
Comment (macrozone) [reply client/router.js]: also my prettier. Damn.

File: client/router.js (hunk 4 above)
Comment (macrozone) [reply client/router.js]: see above. will undo.

File: client/router.js (hunk 5 above)
Comment (macrozone) [reply client/router.js]: can do that.

File: client/router.js (hunk 3 above)
Comment (dr-dimitru) [reply client/router.js]: This ES6 I like, right way to do so, this lib was ES5, so I haven't changed all the code (yet)

**Comment by macrozone** [comment]
@dr-dimitru can you take a look again?

File: docs/api/initialize.md [hunk 6]
```diff
@@ -14,7 +15,11 @@ So, that's where `FlowRouter.wait()` comes to save you. You need to call it dire
 You can pass these options to initialize:
 
 - `hashbang` (defaults to `false`): Enable hashbang urls like `mydomain.com/#!/mypath`.
-- `disableLinkDetection` (defaults to `false`): When true, `<a href>` tags in your app won't automatically call flow router and will do the browser's default page load instead. This is identical to how `react-router` behaves. You can create a `<Link />` component that calls `FlowRouter.go` in its `onClick` handler. This way, you have more control over your links.
+- `page`: Options for [page.js](https://github.com/visionmedia/page.js)
```
Comment (dr-dimitru) [inline docs/api/initialize.md]: Better to point user to the exact point of [`page([options])`](https://github.com/visionmedia/page.js#pageoptions): 
```
https://github.com/visionmedia/page.js#pageoptions
```

File: docs/api/initialize.md [hunk 7]
```diff
@@ -1,14 +1,27 @@
 ### initialize method
 
 ```js
-FlowRouter.initialize();
+FlowRouter.initialize(options);
 ```
+ - `options` {*Object*}
```
Comment (dr-dimitru) [inline docs/api/initialize.md]: We should have a list of nested options here as subitems or anchor link to options header. I prefer a list, like:
```
  - `options` {*Object*}
    - `hashbang` {*Boolean*} - Enable hashbang urls like `mydomain.com/#!/mypath`, default: `false`
    - `page` {*Object*} - Options for [page.js](https://github.com/visionmedia/page.js)
       - `link` {*Boolean*} - When false, `<a href>` tags in your app won't automatically call flow router and will do the browser's default page load instead. This is identical to how `react-router` behaves. You can create a `<Link />` component that calls `FlowRouter.go` in its `onClick` handler. This way, you have more control over your links. Default: `true`
       - Other options can be found in a [`page.js` docs](https://github.com/visionmedia/page.js#pageoptions)
```

Just want to follow the same styling across the all docs

**Review by dr-dimitru — CHANGES_REQUESTED** [review]
Left more comments.
And we should consider accept most of recommendations from your linter, but not one-liner functions 😬

**Comment by macrozone** [comment]
@dr-dimitru ok, added your suggestions

**Comment by macrozone** [comment]
@dr-dimitru i usually use prettier. prettier probably would not have done this one liner ;-) i turned it off, but lint-fix was still active and did this formatting with my local rules (which is more or less eslint-airbnb) ;-)

**Comment by dr-dimitru** [comment]
@macrozone we try to follow [meteor's code style guidelines](https://guide.meteor.com/code-style.html)

**Review by dr-dimitru — APPROVED** [review]

**Comment by dr-dimitru** [comment]
Going to test on my end and publish later tonight

**Comment by macrozone** [comment]
@dr-dimitru could you already publish or need further testing?

**Comment by dr-dimitru** [comment]
Need few more minutes, and will be published

**Comment by dr-dimitru** [comment]
Published as [`v3.5.1`](https://github.com/VeliovGroup/flow-router/releases/tag/3.5.1)

**Pull request merged**